third_party/ngen/ngen_emulation.hpp

```diff
@@ -229,7 +229,7 @@ struct EmulationImplementation {
                                         : i * rd.getHS();
         int noffset = rd.getOffset() + advance;
         if (noffset >= ne) {
-            noffset--;
+            noffset = advance - (noffset - ne);
```

Comment:
This doesn't look right, shouldn't it be:
```suggestion
            noffset -= ne;
```